sql/core/src/main/scala/org/apache/spark/sql/execution/arrow/ArrowWriter.scala

```diff
@@ -80,9 +81,7 @@ object ArrowWriter {
 
 class ArrowWriter(val root: VectorSchemaRoot, fields: Array[ArrowFieldWriter]) {
 
-  def schema: StructType = StructType(fields.map { f =>
-    StructField(f.name, f.dataType, f.nullable)
-  })
+  def schema: StructType = ArrowUtils.fromArrowSchema(root.getSchema())
```

Comment:
As `NullVector` has a fixed field name `"$data$"` for some reason, we have to refer the schema from `VectorSchemaRoot` instead.

Reply:
This shouldn't be the case. I made https://issues.apache.org/jira/browse/ARROW-11382 to be able to change the field name. It's probably better to create from the schema as you did here anyway though.